### PR TITLE
Rails 4.2 support

### PR DESCRIPTION
**Pull request by lucasmazza**
We still have a broken test due escaping, probably similar to what is going on with the Simple Form test suite (https://github.com/plataformatec/simple_form/issues/1130). Beside that, I think the `raise_on_spam` flag could use some better naming and some docs :smile:.

### CODE REVIEW

**Comment by lucasmazza** [comment]
I'm picking this one back and finishing it this week :pray:.

**Comment by lucasmazza** [comment]
Reworked this one to avoid the railtie code and run the suite against the latest 4.2 release.

The only issue left so far is that I tried to use the `deliver_later` API but ActiveJob doesn't know how to serialize a `MailForm` instance, and we will get failures like `ActiveJob::SerializationError: Unsupported argument type: ContactForm`.

**Comment by lucasmazza** [comment]
After some spikes on attempting to support some sort of serialization to integrate `mail_form` with AJ/GlobalID I believe this won't be an easy thing to fix. We could ship a version using `deliver_now` and work on this later.

/cc @carlosantoniodasilva @josevalim @rafaelfranca

**Comment by rafaelfranca** [comment]
Makes sense. Lets release a version to get people using it with rails 4.2
and after we can think in a way to support delayed delivery

File: test/test_helper.rb [hunk 1]
```diff
@@ -11,8 +15,11 @@
 ActionMailer::Base.delivery_method = :test
 I18n.enforce_available_locales = false
 
-$:.unshift File.dirname(__FILE__) + '/../lib'
-require 'mail_form'
+module Rails
+  def self.env
+    @env ||= ActiveSupport::StringInquirer.new('test')
+  end
+end
```
Comment (carlosantoniodasilva) [inline test/test_helper.rb]: Are we using Rails.env somewhere?

**Comment by carlosantoniodasilva** [comment]
Shipping with `deliver_now` seems fine to me, since this is pretty much how it works right now. :+1:

File: test/test_helper.rb (hunk 1 above)
Comment (lucasmazza) [inline test/test_helper.rb]: Yes, we have a `Rails.env.development?` check - what changed with 4.2 is that now the `Rails` module is defined by the html sanitization gem, and we get a `NoMethodError` in the tests. We could throw a `respond_to?` check in the gem code but I rather leave this setup mismatch fix in the test suite itself.

Comment (carlosantoniodasilva) [inline test/test_helper.rb]: Sounds reasonable, I remember some issues with the gem defining the Rails namespace ¯\_(ツ)_/¯.

**Pull request merged**